From d2d10ee33d9a921131e7d9d4ad76d6b178014970 Mon Sep 17 00:00:00 2001
From: Erez Shaul <eshaul@habana.ai>
Date: Thu, 05 Sep 2024 16:24:37 +0300
Subject: [PATCH] [FSW-9045] fs1: use addresses and sizes from emb-specs

most of the addresses & sizes from fs1 common.h exists
in embedded specs - address_space.h
therefore it is preferred to use them from common code.

Change-Id: I4fcea8ac7c7fd2bfccb9a3d9fc22cf24ae0ea1aa
Signed-off-by: Erez Shaul <eshaul@habana.ai>
---

diff --git a/arch/fs1/arc/common.c b/arch/fs1/arc/common.c
index 008fb1a..b518518 100644
--- a/arch/fs1/arc/common.c
+++ b/arch/fs1/arc/common.c
@@ -47,6 +47,7 @@
 #include <pll.h>
 #include <efuse.h>
 #include <boot_desc.h>
+#include <address_space.h>
 #include "common.h"
 #include "hl.h"
 #include "auto_gen_pcie_phy_csr_locations.h"
@@ -258,8 +259,8 @@
 
 static u32 fs1_translate_arc_addr(u64 addr)
 {
-	if (addr >= HL_SP_SRAM_BASE && addr < (HL_SP_SRAM_BASE + (u64)HL_SP_SRAM_SIZE)) {
-		addr = (addr & SPSRAM_ADDR_MASK) + SP_SRAM_BASE;
+	if (addr >= IO0_SP_SRAM_FULL_BASE && addr < (IO0_SP_SRAM_FULL_BASE + (u64)IO_SP_SRAM_SIZE)) {
+		addr = (addr & SPSRAM_ADDR_MASK) + IO0_SP_SRAM_BASE;
 		goto ret_addr;
 	}
 
@@ -285,8 +286,8 @@
 
 static enum dma_dest_type preboot_get_dma_dest_type(uint64_t addr)
 {
-	if (addr >= HL_FW_SRAM_FULL_BASE &&
-	    addr < HL_FW_SRAM_FULL_BASE + HL_FW_SRAM_SIZE)
+	if (addr >= IO0_FW_SRAM_FULL_BASE &&
+	    addr < IO0_FW_SRAM_FULL_BASE + IO_FW_SRAM_SIZE)
 		return DMA_DEST_SRAM;
 
 	return DMA_DEST_HBW;
@@ -521,17 +522,17 @@
 // based on lkd_dev_spec_get_img_addr from preboot
 static u64 get_64bit_addr_from_32bit_addr(uintptr_t base)
 {
-	if (base >= SP_SRAM_BASE && base < SP_SRAM_BASE + HL_SP_SRAM_SIZE)
-		return (base & ~SP_SRAM_BASE) + HL_SP_SRAM_BASE;
+	if (base >= IO0_SP_SRAM_BASE && base < IO0_SP_SRAM_BASE + IO_SP_SRAM_SIZE)
+		return (base & ~IO0_SP_SRAM_BASE) + IO0_SP_SRAM_FULL_BASE;
 
-	if (base >= FW_SRAM_BASE && base < FW_SRAM_BASE + HL_FW_SRAM_SIZE)
-		return (base & ~FW_SRAM_BASE) + HL_FW_SRAM_FULL_BASE;
+	if (base >= IO0_FW_SRAM_BASE && base < IO0_FW_SRAM_BASE + IO_FW_SRAM_SIZE)
+		return (base & ~IO0_FW_SRAM_BASE) + IO0_FW_SRAM_FULL_BASE;
 
-	if (base >= ARC_MNG_ICCM_BASE && base < ARC_MNG_ICCM_BASE + ARC_MNG_ICCM_SIZE)
-		return (base & ~ARC_MNG_ICCM_BASE) + ARC_MNG_ICCM_FULL_BASE;
+	if (base >= IO0_MNG_ICCM_BASE && base < IO0_MNG_ICCM_BASE + IO_MNG_ICCM_SIZE)
+		return (base & ~IO0_MNG_ICCM_BASE) + IO0_MNG_ICCM_FULL_BASE;
 
-	if (base >= 0 && base < ARC_MNG_ICCM_SIZE)
-		return (base & ~ARC_MNG_ICCM_BASE) + ARC_MNG_ICCM_FULL_BASE;
+	if (base >= 0 && base < IO_MNG_ICCM_SIZE)
+		return (base & ~IO0_MNG_ICCM_BASE) + IO0_MNG_ICCM_FULL_BASE;
 
 	preboot_puts_raw(__func__);
 	preboot_puts(": error - unsupported addr\n");
@@ -548,11 +549,11 @@
 	uint32_t sram_loc_addr = ALIGN_TO_16_BYTES(PPBOOT_LAST_ADDR);
 	const uint32_t *csr_loc_data = get_csr_locations();
 
-	// Copy SRAM_LOCATIONS to PCIE_FW_SRAM_BASE
+	// Copy SRAM_LOCATIONS to IO0_PCIE_FW_SRAM_FULL_BASE
 	fs1_mem_copy(get_64bit_addr_from_32bit_addr(sram_loc_addr),
-		    HL_PCIE_FW_SRAM_BASE_FULL,
+		    IO0_PCIE_FW_SRAM_FULL_BASE,
 		    (uint32_t)PHY_FW_SIZE, false);
 
-	//Copy CSR_LOC_DATA to PCIE_FW_CSR_BASE
-	memcpy((void *)PCIE_FW_CSR_BASE, (const void *)csr_loc_data, get_csr_locations_size());
+	//Copy CSR_LOC_DATA to IO0_PCIE_FW_CSR_BASE
+	memcpy((void *)IO0_PCIE_FW_CSR_BASE, (const void *)csr_loc_data, get_csr_locations_size());
 }
diff --git a/arch/fs1/arc/common.h b/arch/fs1/arc/common.h
index d1a3bb7..09a32d7 100644
--- a/arch/fs1/arc/common.h
+++ b/arch/fs1/arc/common.h
@@ -12,49 +12,16 @@
 
 #define HL_FW_BOOT_PTRS_BASE		0
 
+/* same as IO0_SP_SRAM_FULL_BASE (left here for backwards compatibility) */
 #define HL_SP_SRAM_BASE			0x300007FF8A00000ULL
 #define HL_SP_SRAM_SIZE			SZ_1M
-#define HL_FW_SRAM_FULL_BASE		0x300007FF89A0000ULL
-#define HL_FW_SRAM_SIZE			SZ_128K
-#define HL_FW_MEM_BASE_FULL		0x300007FF8800000ULL
-#define HL_FW_MEM_SIZE			SZ_256M
-#define HL_CFG_MEM_BASE_FULL		0x300007FF8000000ULL
-#define HL_CFG_MEM_SIZE			SZ_64M
-#define HL_PCIE_FW_SRAM_BASE_FULL	0x300007FF8B80000ULL
-
-#define ARC_PID_MEM_SIZE		(ARC_PID_ICCM_SIZE + ARC_PID_DCCM_SIZE)
-#define ARC_MNG_MEM_SIZE		(ARC_MNG_ICCM_SIZE + ARC_MNG_DCCM_SIZE)
-
-#define ARC_RST_ICCM_FULL_BASE		0x300007FF8800000ull
-#define ARC_RST_ICCM_SIZE		SZ_64K
-#define ARC_RST_DCCM_FULL_BASE		0x300007FF8810000ull
-#define ARC_RST_DCCM_SIZE		SZ_32K
-#define ARC_MNG_ICCM_FULL_BASE		0x300007FF8840000ull
-#define ARC_MNG_ICCM_SIZE		SZ_256K
-#define ARC_MNG_DCCM_FULL_BASE		0x300007FF8880000ull
-#define ARC_MNG_DCCM_SIZE		SZ_128K
-#define ARC_PID_ICCM_FULL_BASE		0x300007FF88C0000ull
-#define ARC_PID_ICCM_SIZE		SZ_128K
-#define ARC_PID_DCCM_FULL_BASE		0x300007FF88E0000ull
-#define ARC_PID_DCCM_SIZE		SZ_64K
-#define ARC_MISC_ICCM_FULL_BASE		0x300007FFA900000ull
-#define ARC_MISC_ICCM_SIZE		SZ_256K
-#define ARC_MISC_DCCM_FULL_BASE		0x300007FFA940000ull
-#define ARC_MISC_DCCM_SIZE		SZ_64K
 
 /* CFG is at Arc region F, specs CFG base address is 0x70000000. 7+8 = 0xF */
+/* defined here in addition to specs because they are in use im common code */
 #define HL_CFG_BASE			0x80000000
 #define FW_MEM_REGION_BASE		0x78800000 /* IO Die0 Half0 FW memories */
-#define PCIE_FW_SRAM_BASE		(HL_CFG_BASE + mmIO_0_CFG_HALF_0_FW_MEM_PCIE_FW_SRAM_BASE +\
-					 mmPCIE_FW_SRAM_SRAM_LOCATIONS)
-#define PCIE_FW_CSR_BASE		(HL_CFG_BASE + mmIO_0_CFG_HALF_0_FW_MEM_PCIE_FW_SRAM_BASE +\
-					 mmPCIE_FW_SRAM_CSR_LOCATIONS)
 #define FW_SRAM_BASE			(FW_MEM_REGION_BASE + 0x1A0000)
 #define SP_SRAM_BASE			(FW_MEM_REGION_BASE + 0x200000)
-#define ARC_MNG_ICCM_BASE		(FW_MEM_REGION_BASE + 0x40000)
-#define ARC_MNG_DCCM_BASE		(FW_MEM_REGION_BASE + 0x80000)
-#define ARC_PID_ICCM_BASE		(FW_MEM_REGION_BASE + 0xC0000)
-#define HL_SPI_BASE			0x94000000
 
 #define BOOT_INFO_BASE			SP_SRAM_BASE
 #define BOOT_INFO_SIZE			SZ_32K
diff --git a/arch/fs1/arc/palad.c b/arch/fs1/arc/palad.c
index 0b47494..7e324ba 100644
--- a/arch/fs1/arc/palad.c
+++ b/arch/fs1/arc/palad.c
@@ -23,6 +23,8 @@
 #include <efuse.h>
 #include <hl.h>
 
+#include <address_space.h>
+
 #define SYS_CLOCK_FREQUENCY	250000000ULL /* PCI PLL DIV3 Emulation clk */
 
 #define UART_CLOCK_FREQUENCY	SYS_CLOCK_FREQUENCY
@@ -240,7 +242,7 @@
 	 * exceeds MISC ICCM size (256kb), the rest is stored in MISC DCCM.
 	 * An efuse bit is controlling whether the application binary is larger than MISC ICCM size.
 	 */
-	fs1_mem_copy(ARC_MISC_ICCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL, ARC_MISC_ICCM_SIZE, false);
+	fs1_mem_copy(IO0_MISC_ICCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL, IO_MISC_ICCM_SIZE, false);
 
 	err = preboot_fuse_read(IO_EFUSE_PLDM_CFG_0_OFFSET, &val);
 	if (err)
@@ -251,8 +253,8 @@
 	if (!val)
 		return;
 
-	fs1_mem_copy(ARC_MISC_DCCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL + ARC_MISC_ICCM_SIZE,
-		     ARC_MISC_DCCM_SIZE, false);
+	fs1_mem_copy(IO0_MISC_DCCM_FULL_BASE, PREBOOT_TEXT_BASE_FULL + IO_MISC_ICCM_SIZE,
+		     IO_MISC_DCCM_SIZE, false);
 }
 
 static void btldr_copy_bootinfo(void)
@@ -276,7 +278,7 @@
 	 * Since SPSRAM content is erazed on cold reboot, fresh copy stored in MISC DCCM and copied
 	 * back to the SPSRAM as part of the preparation to run preboot rtos.
 	 */
-	fs1_mem_copy(ARC_MISC_DCCM_FULL_BASE, HL_SP_SRAM_BASE, BOOT_INFO_SIZE, false);
+	fs1_mem_copy(IO0_MISC_DCCM_FULL_BASE, IO0_SP_SRAM_FULL_BASE, BOOT_INFO_SIZE, false);
 }
 
 void btldr_copy_data(void)
